localstack/constants.py

```diff
@@ -199,7 +199,9 @@
 DEFAULT_DEVELOP_PORT = 5678
 
 # Default bucket name of the s3 bucket used for local lambda development
-DEFAULT_BUCKET_MARKER_LOCAL = "__local__"
+# This name should be accepted by all IaC tools, so should respect s3 bucket naming conventions
+DEFAULT_BUCKET_MARKER_LOCAL = "hot-reloading-bucket"
+OLD_DEFAULT_BUCKET_MARKER_LOCAL = "__local__"
```

Comment:
```suggestion
LEGACY_DEFAULT_BUCKET_MARKER_LOCAL = "__local__"
```

Would be a bit more inline with the conventions in config.py for example